File: my-submissions/m1302.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        def helper(curr: Optional[TreeNode]) -> (int, int) : # (sum, height)
            if not curr.left and not curr.right:
                return (curr.val, 1)
            if not curr.left :
                right = helper(curr.right)
                return (right[0], right[1] + 1)
            if not curr.right :
                left = helper(curr.left)
                return (left[0], left[1] + 1)

            left = helper(curr.left)
            right = helper(curr.right)

            if left[1] == right[1] :
                return (left[0] + right[0], left[1] + 1)
            if left[1] > right[1] :
                return (left[0], left[1] + 1)
            return (right[0], right[1] + 1)

        return helper(root)[0]

File: my-submissions/m1302.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var deepestLeavesSum = function(root) {
    var deepestDepth = 0;
    var output = 0;
    
    function dfs(curr, currDepth) {
        if (curr == undefined) {
            return;
        }
        if (currDepth > deepestDepth) {
            deepestDepth = currDepth;
            output = 0;
        }
        if (currDepth == deepestDepth) {
            output += curr.val;
        }
        currDepth++;
        dfs(curr.left, currDepth)
        dfs(curr.right, currDepth)
    }
    dfs(root, 0);
    return output;
};